lib/Doctrine/Common/DataFixtures/ReferenceRepository.php

```diff
@@ -77,8 +77,24 @@ protected function getIdentifier($reference, $uow)
         // In case Reference is not yet managed in UnitOfWork
         if ( ! $uow->isInIdentityMap($reference)) {
             $class = $this->manager->getClassMetadata(get_class($reference));
-
-            return $class->getIdentifierValues($reference);
+            
+            $values = $class->getIdentifierValues($reference);
+
+            // See https://github.com/doctrine/data-fixtures/issues/135 and
+            // https://github.com/doctrine/data-fixtures/issues/167
+            foreach ($values as $key => $value) {
+                if (!is_scalar($value)) {
+                    // To prevent "Array to string conversion" in
+                    // UnitOfWork::tryGetById, we can only return a single
+                    // identifier. This does mean that related entities that
+                    // themselves have a composite primary key are unsupported.
+                    $proxyId = $this->getIdentifier($value, $uow);
+                    $keys = array_keys($proxyId);
+                    $values[$key] = $proxyId[$keys[0]];
```

Comment:
In most cases, where the related entity has a single (not composite) primary key, `count($proxyId) === 1`.

Imploding the keys is an option, but unfortunately won’t help much. When the references are unserialized, references are [retrieved from the entity manager](https://github.com/doctrine/data-fixtures/blob/master/lib/Doctrine/Common/DataFixtures/ProxyReferenceRepository.php#L93). However, Doctrine does not support something like:

```php
$roleRef = $em->getReference(self::TEST_ENTITY_USER_ROLE, [
    'role' => 1,
    'user' => [
        'id' => 1,
        'code' => '007'
    ]
]);

```